src/python/pants/backend/go/util_rules/external_module.py

```diff
@@ -322,10 +322,14 @@ async def compute_package_import_paths_from_external_module(
             ),
         ),
     )
-    return ExternalModulePkgImportPaths(
-        metadata["ImportPath"]
-        for metadata in ijson.items(json_result.stdout, "", multiple_values=True)
-    )
+
+    if json_result.stdout:
+        return ExternalModulePkgImportPaths(
+            metadata["ImportPath"]
+            for metadata in ijson.items(json_result.stdout, "", multiple_values=True)
+        )
+    else:
+        return ExternalModulePkgImportPaths()
```

Comment:
Per the style guide, prefer early returns: https://www.pantsbuild.org/docs/style-guide#prefer-early-returns-in-functions

```suggestion
    if not json_result.stdout:
        return ExternalModulePkgImportPaths()
    return ExternalModulePkgImportPaths(
        metadata["ImportPath"]
        for metadata in ijson.items(json_result.stdout, "", multiple_values=True)
    )
```